### ISSUE TITLE
loadMetadata can't return immediately after all the metadata is loaded

### ISSUE THREAD

**Issue by TrafalgarZZZ** [Alluxio/alluxio#12126]
**Alluxio Version:**
What version of Alluxio are you using?
https://github.com/Alluxio/alluxio/tree/branch-2.3-fuse

**Describe the bug**
I mount an Aliyun OSS including about 3.8 million small files on Alluxio, after then I run the `alluxio fs loadMetadata -R /` to load metadata of all the files in my OSS. 
What confuses me is that before the `alluxio fs loadMetadata -R /` return, I run the followiing command:
```
bash-4.4# alluxio fsadmin report metrics | grep Master.TotalPaths
Master.TotalPaths  (Type: GAUGE, Value: 3,804,850)
```
And the value "3804850" is exactly the number of all my files in my OSS. So it seems to me all the metadata has already been loaded, however, the `alluxio fs loadMetadata -R /` command still won't return until about 10min later.

**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)
Try to load metadata of large amount of files and run the `report metrics` to check `Master.TotalPaths` 

**Expected behavior**
The loadMetadata command should return as soon as the job is done

**Urgency**
Describe the impact and urgency of the bug.

**Additional context**

**Comment by apc999** [comment]
this may be related to client-side calculating results streamed from master
We may need bring back some previous change from 
https://github.com/Alluxio/alluxio/pull/3272/files
https://github.com/Alluxio/alluxio/pull/8546/files

**Comment by apc999** [comment]
this should be fixed by #12421
